### PR TITLE
Adding BoundingBoxVectorPostProcessor

### PR DESCRIPTION
**Pull request by permcody**
Adding a Postprocessor to output bounding boxes for all features.

### CODE REVIEW

**Comment by permcody** [comment]
Tag @aeslaughter - see the CSV file included in this PR. Note that the feature ID may be duplicated. This isn't super important to you, it just shows you that some features have more than one bounding box (when periodic boundary conditions are applied). Later we can add additional columns for color, etc.

File: modules/phase_field/include/vectorpostprocessors/FeatureBoundingBoxVectorPostprocessor.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  virtual void initialize() override;
+  virtual void execute() override;
+  virtual void finalize() override;
+
+protected:
+  /// A reference to the feature flood count object
+  const FeatureFloodCount & _feature_counter;
+
+  VectorPostprocessorValue & _feature_id;
```
Comment (aeslaughter) [inline modules/phase_field/include/vectorpostprocessors/FeatureBoundingBoxVectorPostprocessor.h]: This should get a /// comment

File: modules/phase_field/include/vectorpostprocessors/FeatureBoundingBoxVectorPostprocessor.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  VectorPostprocessorValue & _min_x;
+  VectorPostprocessorValue & _min_y;
+  VectorPostprocessorValue & _min_z;
+  VectorPostprocessorValue & _max_x;
+  VectorPostprocessorValue & _max_y;
+  VectorPostprocessorValue & _max_z;
```
Comment (aeslaughter) [inline modules/phase_field/include/vectorpostprocessors/FeatureBoundingBoxVectorPostprocessor.h]: You should wrap these all in a doxygen comment

///@{
/// Min, max ....
VectorPostprocessorValue ...
///@}

File: modules/phase_field/include/vectorpostprocessors/FeatureBoundingBoxVectorPostprocessor.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+private:
+  /// Add volume contributions to one or entries in the feature volume vector
+  void accumulateVolumes(const Elem * elem, const std::vector<unsigned int> & var_to_features, std::size_t num_features);
+
+  /// Calculate the integral value of the passed in variable (index)
+  Real computeIntegral(std::size_t var_index) const;
+
+  const std::vector<MooseVariable *> & _vars;
```
Comment (aeslaughter) [inline modules/phase_field/include/vectorpostprocessors/FeatureBoundingBoxVectorPostprocessor.h]: docs

File: modules/phase_field/src/vectorpostprocessors/FeatureBoundingBoxVectorPostprocessor.C [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+#include "MooseMesh.h"
+#include "Assembly.h"
+
+#include "libmesh/quadrature.h"
+
+template<>
+InputParameters validParams<FeatureBoundingBoxVectorPostprocessor>()
+{
+  InputParameters params = validParams<GeneralVectorPostprocessor>();
+
```
Comment (aeslaughter) [inline modules/phase_field/src/vectorpostprocessors/FeatureBoundingBoxVectorPostprocessor.C]: addClassDescription

**Review by aeslaughter — CHANGES_REQUESTED** [review]

**Comment by aeslaughter** [comment]
@permcody Do you still want me to work on this capability?

**Comment by permcody** [comment]
Maybe... Clearly it's not a high priority for me at the moment since this PR is still sitting here.

**Comment by permcody** [comment]
@dschwen - Any interest in me fixing up this PR? I thought I wanted it for visualization but never got around to using it. It could go in with minimal effort but I don't know if anyone actually cares. It might just get killed.

**Comment by aeslaughter** [comment]
@permcody You don't seem to need this anymore, we can re-visit when needed again.

**Comment by aeslaughter** [comment]
@permcody You don't seem to need this anymore, we can re-visit when needed again.

**Pull request closed without merging**